lib/MusicBrainz/Server/Controller/EditNote.pm

```diff
@@ -0,0 +1,208 @@
+package MusicBrainz::Server::Controller::EditNote;
+use Moose;
+use MooseX::MethodAttributes;
+use namespace::autoclean;
+
+use List::AllUtils qw( first );
+use MusicBrainz::Server::Constants qw( $CONTACT_URL );
+use MusicBrainz::Server::Data::Utils qw( load_everything_for_edits );
+use MusicBrainz::Server::Validation qw( is_database_row_id );
+use MusicBrainz::Server::Translation qw( l );
+
+extends 'MusicBrainz::Server::Controller';
+with 'MusicBrainz::Server::Controller::Role::Load' => {
+    model => 'EditNote',
+    entity_name => 'edit_note',
+};
+
+sub base : Chained('/') PathPart('edit-note') CaptureArgs(0) { }
+
+sub _load
+{
+    my ($self, $c, $edit_note_id) = @_;
+    return unless is_database_row_id($edit_note_id);
+    my $edit_note = $c->model('EditNote')->get_by_id($edit_note_id);
+    $c->model('Editor')->load($edit_note);
+    $c->model('EditNoteChange')->load_latest($edit_note);
+    return $edit_note;
+}
+
+sub detach_if_cannot_modify {
+    my ($self, $c, $edit_note, $edit) = @_;
+
+    my $is_own_note = $c->user && $c->user->id == $edit_note->{editor_id};
+    my $is_admin = $c->user && $c->user->is_account_admin;
+
+    unless ($is_own_note || $is_admin) {
+      $c->stash(
+          component_path  => 'user/UserMessage',
+          component_props => {
+              title    => l('Cannot modify edit note'),
+              message  => l('You cannot modify other users’ edit notes'),
+          },
+          current_view    => 'Node',
+      );
+      $c->detach;
+    }
+
+    my $is_note_deleted = $edit_note->{latest_change} &&
+                          $edit_note->{latest_change}->{status} eq 'deleted';
+
+    if ($is_note_deleted && !$is_admin) {
+      $c->stash(
+          component_path  => 'user/UserMessage',
+          component_props => {
+              title    => l('Cannot modify edit note'),
+              message  => l('This note has already been deleted.'),
+          },
+          current_view    => 'Node',
+      );
+      $c->detach;
+    }
+
+    my $all_edit_notes = $edit->{edit_notes};
+    my $note_index = first {
+      @$all_edit_notes[$_]->id == $edit_note->id
+    } 0..@$all_edit_notes;
```

Comment:
This might be handy here https://metacpan.org/pod/List::AllUtils#first_index-BLOCK-LIST
```suggestion
    my $note_index = first_index {
      $all_edit_notes->[$_]->id == $edit_note->id
    } @$all_edit_notes;
```

Reply:
Guess you meant just `$_->id == $edit_note->id` for the second line?